src/gpio.rs

```diff
@@ -761,6 +761,46 @@ macro_rules! gpio {
                     }
                 }
 
+                impl IoPin<Self, Self>
+                for $PXi<Output<OpenDrain>>
+                {
+                    type Error = Never;
+                    fn into_input_pin(self) -> Result<Self, Self::Error> {
+                        Ok(self)
+                    }
+                    fn into_output_pin(mut self, state: PinState) -> Result<Self, Self::Error> {
+                        self.set_state(state)?;
```

Comment:
I see here the problem that if setting the state fails, the pin is dropped.

Reply:
The `set_low` and `set_high` methods implemented here are infallible (use the `Never` type as an error), so [`set_state`](https://docs.rs/embedded-hal/0.2.6/embedded_hal/digital/v2/trait.OutputPin.html#method.set_state)  would also be infallible. 

Maybe we should implement our own infallible `set_state` (as [stm32f4xx-hal does](https://github.com/stm32-rs/stm32f4xx-hal/blob/f80106fa9e41c2180cd694d3b17b97198c7d4d99/src/gpio.rs#L431)) and use that?